src/Middleware/Diagnostics/src/ExceptionHandler/ExceptionHandlerMiddlewareImpl.cs

```diff
@@ -188,7 +188,12 @@ private async Task HandleException(HttpContext context, ExceptionDispatchInfo ed
                 }
                 else
                 {
-                    handled = await _problemDetailsService!.TryWriteAsync(new()
+                    if (_problemDetailsService is null)
+                    {
+                        throw new InvalidOperationException(Resources.ExceptionHandlerOptions_NotConfiguredCorrectly);
+                    }
+
+                    handled = await _problemDetailsService.TryWriteAsync(new()
```

Comment:
There's still an issue if this returns false. It will fall through to the IOE on line 217/222 and give the incorrect error about a 404. I think if problem details can't handle the error then we need to at least produce a different error.